caravel/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -29,30 +30,34 @@ class ChartContainer extends React.Component {
   }
 
   componentWillMount() {
-    this.setState({ mockSlice: this.getMockedSliceObject() });
+    this.setState({ mockSlice: this.getMockedSliceObject(this.props) });
   }
 
   componentDidMount() {
     this.renderVis();
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
The reason I added this was I noticed componentWillMount is not called at props change. But we do need to reload mock slice both before render and at prop change.